node_groups.tf

```diff
@@ -19,6 +19,20 @@ locals {
   }
 }
 
+resource "time_sleep" "this" {
+  count = var.create ? 1 : 0
+
+  create_duration = var.dataplane_wait_duration
+
+  triggers = {
+    cluster_name     = aws_eks_cluster.this[0].name
```

Comment:
I understand this now, but maybe add a comment somewhere near or inside of this resource for the next review when we will forget the whole story.

The reasoning and the solution are neat but not very obvious. :)

Reply:
updated in https://github.com/terraform-aws-modules/terraform-aws-eks/pull/2478/commits/99a837c324bfedab8f686ea3cb94410ae2c5d550